### PR TITLE
Shocking Messages Over CAN

### PR DESCRIPTION
**Pull request by AarC10**

### CODE REVIEW

File: include/HUDL/HUDL.hpp [hunk 1]
```diff
@@ -30,37 +38,285 @@ class HUDL {
...
+        {
+            .Key = CO_KEY(0x1602, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 1,
+        },
+        {
+            .Key = CO_KEY(0x1602, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = CO_LINK(0x2101, 0, 8),
+        },
```
Comment (mjh9585) [inline include/HUDL/HUDL.hpp]: Same thing as pull request #3, check your link sizes.

File: include/HUDL/HUDL.hpp [hunk 2]
```diff
@@ -30,37 +38,285 @@ class HUDL {
...
+         */
+        {
+            .Key = CO_KEY(0x1402, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 3,
+        },
+        {
+            .Key = CO_KEY(0x1402, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 0x40000180 + 0x05,
```
Comment (mjh9585) [inline include/HUDL/HUDL.hpp]: Why is this hard coded like this?

File: include/HUDL/HUDL.hpp [hunk 3]
```diff
@@ -13,14 +16,19 @@ namespace HUDL {
...
+* The node ID used to identify the device on the CAN network.
+*/
+    static constexpr uint8_t NODE_ID = 0x11;
+
+    /**
+     * Default Constructor for the HUDL class
+     *
+     * @param[in] reg_select is the register select pin
+     * @param[in] reset is the reset pin
+     * @param[in] cs is the chip select piarrayn
```
Comment (mjh9585) [inline include/HUDL/HUDL.hpp]: update documentation with the removal of the CS pin.

**Review by mjh9585 — CHANGES_REQUESTED** [review]
Hard to tell how many of these changes are from #3  to avoid commenting on the same things, but overall looks good. mostly just the same fixes need to be added to the object dictionary.

**Comment by mjmagee991** [comment]
[HUDL CAN](https://trello.com/c/vCpuUwvL/39-hudl-can)

File: include/HUDL/HUDL.hpp [hunk 4]
```diff
@@ -248,6 +270,22 @@ class HUDL {
             .Data = CO_LINK(0x2100, 3, 32),
         },
 
+        /**
+         * RPDO2 mapping, determines the PDO messages to send when RPDO1 is triggered
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
         * RPDO2 mapping, determines the PDO messages to receive when RPDO2 is triggered
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Just a typo in the documentation

File: include/HUDL/HUDL.hpp (hunk 4 above)
Comment (mjh9585) [reply include/HUDL/HUDL.hpp]: Should also be "message to _receive_" instead of send.

**Review by mjh9585 — APPROVED** [review]
Other than Matt's comment looks good. One note I will make is that I believe there are two battery packs on the bike. Not sure how we want to handle this since they all have different IDs.

**Comment by mjmagee991** [comment]
We still haven't decided how to handle the fact that there are multiple packs, but it should be fairly trivial to update this once we figure that out.

**Review by mjmagee991 — APPROVED** [review]
Looks great

**Pull request merged**